packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java

```diff
@@ -85,11 +85,92 @@ public void onActivityDestroyed(Activity activity) {
     @Override
     public void onActivityStopped(Activity activity) {
       if (thisActivity == activity) {
-        delegate.saveStateBeforeResult();
+        activityState.getDelegate().saveStateBeforeResult();
       }
     }
   }
 
+  private class ActivityState {
+    private Application application;
+    private Activity activity;
+    private ImagePickerDelegate delegate;
+    private MethodChannel channel;
+    private LifeCycleObserver observer;
+    private ActivityPluginBinding activityBinding;
+
+    // This is null when not using v2 embedding;
+    private Lifecycle lifecycle;
+
+    ActivityState(
```

Comment:
needs java doc

Reply:
the behavior of `ActivityState#release()` depends upon which constructor is used.

Reply:
done.